### ISSUE TITLE
[service] Configure tools.apple:sdk_path for MesonToolchain

### ISSUE THREAD

**Issue by jwillikers** [conan-io/conan-center-index#11613]
Would it be possible to configure the `tools.apple:sdk_path` in Conan Center so that the MesonToolchain can be used? It requires this to be set [here](https://github.com/conan-io/conan/blob/3507c0978142409d6f13caf967d05c76bcb3b96f/conan/tools/meson/toolchain.py#L171). Both PR #11610 and PR #11053 currently fail to build due to this value not being configured.

**Comment by SSE4** [comment]
@franramirez688 do we need to set it globally in our profiles for Conan Center? or do we need to just enforce it in recipe?
both seems to be a bit inconvinient, couldn't it just work out of the box?

**Comment by franramirez688** [comment]
> @franramirez688 do we need to set it globally in our profiles for Conan Center? or do we need to just enforce it in recipe? both seems to be a bit inconvinient, couldn't it just work out of the box?

Yes, you'd need to set it globally. Now, Conan is getting a more explicit way of working and avoiding (if possible) magic auto-detection so this is part of it. Sorry for any inconvenience!

**Comment by jgsogo** [comment]
Is this something for profiles or global configuration?

**~/.conan/conan.conf**
```ini
[conf]
tools.apple:sdk_path="/.../.../.../..."
``` 

Is the output from `/usr/bin/xcrun -sdk macosx --show-sdk-path` the right value to set there?

**Comment by franramirez688** [comment]
> Is this something for profiles or global configuration?

Yes, both are valid.

> Is the output from `/usr/bin/xcrun -sdk macosx --show-sdk-path` the right value to set there?

Exactly.

**Comment by SSE4** [comment]
@franramirez688 it turns out to be not so easy, and seems to require an additional development
I've checked, and adding a simple "workaround" to the recipe:
```
--- a/recipes/glib/all/conanfile.py
+++ b/recipes/glib/all/conanfile.py
@@ -112,6 +112,9 @@ class GLibConan(ConanFile):
         self.folders.source = self._source_subfolder

     def generate(self):
+        from conan.tools.apple.apple import apple_sdk_path
+        self.conf["tools.apple:sdk_path"] = apple_sdk_path(self)
+
         tc = PkgConfigDeps(self)
         tc.generate()
         tc = AutotoolsDeps(self)
```
it allows to advance build a bit, but it fails eventually:
```
../source_subfolder/meson.build:795:2: ERROR: 'objc' compiler binary not defined in cross or native file
```

so, it seems like we have to configure Objective-C and Objective-C++ compilers as well on Apple platforms in `MesonToolchain`. I would also add some handling for `objc_args`/`objc_link_args`/`objcpp_args `/`objcpp_link_args ` like I proposed before here: https://github.com/conan-io/conan/pull/9711/files#diff-0e1e7a3c21f2c036f65b81f8dc72f539f9c76b2b0a031cd5eca0f1de392bd49eR50-R53
